homework/list.js

```diff
@@ -1,79 +1,95 @@
 'use strict';
 
-(function() {
 
-  var _listNoteContent = [];
-  var _wrapper = document.querySelector('#note-list-wrapper');
+var List = function() {
+    this._listNoteContent = [];
+    this._wrapper = document.querySelector('#note-list-wrapper')
+}
 
-  function start() {
-    fetchList(function(data) {
-      updateList(data);
-      drawList();
-      preloadFirstNote();
-    });
+var ListNote = new List() ;
+
+List.prototype.start=function() {
+    var f = this.updateList.bind(this);
+    
+    this.fetchList()
+        
+        /*.then(function(data){
+        this.updateList(data);
+        }.bind(this))*/
+
+        .then(this.updateList.bind(this))
+        //.then(f)
+        .then(this.drawList.bind(this))
+        .then(this.preloadFirstNote.bind(this));
```

Comment:
Good. You have practiced the `then` in chain. However, you should try how to handle the error with `catch`. And if you do that, please note that the error should be **log AND re-throw**. Since if you only log it, the following chain will still perform some unexpected side-effects and screw your program.